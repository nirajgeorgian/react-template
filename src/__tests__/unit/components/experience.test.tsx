import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Experience from 'components/cards/experience'

const ExperienceCompoent = (
	<Experience
		imagesrc="http://www.thebirdwheel.com/wp-content/uploads/2011/07/New-York-Bike-Portrait-Kim-Burgas-6.jpg"
		pretitle="Experience | Restaurant"
		title="Gazebo"
		city="Abu dhabi"
		country="UAE"
		languages="Urdu,North Indian,leabenese"
		currency="$"
		price="100"
	/>
)
//for rendering
it('renders correctly', () => {
	const { asFragment } = render(ExperienceCompoent)
	expect(asFragment()).toMatchSnapshot()
})

// it('image in the card', () => {
// 	const { getByTestId } = render(ExperienceCompoent)
// 	const ancestor = document.querySelector('[data-testid="stayology_experience_card"]')
// 	const descendant = document.querySelector('[data-testid="satyology_experience_carousel_img"]')
// 	expect(ancestor).toContainElement(descendant)
// })
