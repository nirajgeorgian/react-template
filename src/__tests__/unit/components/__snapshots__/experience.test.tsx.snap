// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-card experience_card ant-card-hoverable"
    data-testid="stayology_experience_card"
  >
    <div
      class="ant-card-cover"
    >
      <img
        alt=""
        data-testid="satyology_experience_carousel_img"
        src="http://www.thebirdwheel.com/wp-content/uploads/2011/07/New-York-Bike-Portrait-Kim-Burgas-6.jpg"
      />
      <div>
        <i
          aria-label="icon: heart"
          class="anticon anticon-heart"
          data-testid="stayology_experience_heart"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="heart"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
            />
          </svg>
        </i>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="ant-row"
      >
        <div
          class="ant-col"
        >
          <span>
            Experience | Restaurant
          </span>
          <div
            class="ant-card-meta"
          >
            <div
              class="ant-card-meta-detail"
            >
              <div
                class="ant-card-meta-title"
              >
                Gazebo
              </div>
            </div>
          </div>
          <span>
            Abu dhabi,UAE
          </span>
        </div>
      </div>
      <div
        class="ant-row"
      >
        <div
          class="ant-col ant-col-12"
          style="justify-content: flex-start;"
        >
          <span>
            Urdu,North Indian,leabenese
            <br />
            <i
              aria-label="icon: star"
              class="anticon anticon-star experience_star"
              style="font-size: 15px;"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="star"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                />
              </svg>
            </i>
            <i
              aria-label="icon: star"
              class="anticon anticon-star"
              style="font-size: 15px;"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="star"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                />
              </svg>
            </i>
            <i
              aria-label="icon: star"
              class="anticon anticon-star"
              style="font-size: 15px;"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="star"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                />
              </svg>
            </i>
            <i
              aria-label="icon: star"
              class="anticon anticon-star"
              style="font-size: 15px;"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="star"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                />
              </svg>
            </i>
            <i
              aria-label="icon: star"
              class="anticon anticon-star"
              style="font-size: 15px;"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="star"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                />
              </svg>
            </i>
          </span>
        </div>
        <div
          class="ant-col ant-col-12"
          style="display: flex; justify-content: flex-end;"
        >
          <span>
            <p>
              <sup>
                <small>
                  $
                </small>
              </sup>
              100
            </p>
            <p
              style="font-size: 10px;"
            >
              Per person
            </p>
            <p />
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
